src/lib.rs

```diff
@@ -2290,8 +2283,19 @@ impl Build {
             None => false,
         };
 
-        let arch = if is_catalyst {
-            match arch {
+        let arch = if is_mac {
+            match arch_str {
+                "i686" => ArchSpec::Device("-m32"),
+                "x86_64" | "aarch64" => ArchSpec::Device("-m64"),
```

Comment:
Needs to handle `x86_64h` here (and look for other places where `arch_str` should handle it).

Reply:
Should be done now, it seems like the original PR predated that target being supported.